coordinator/src/services/translation/request/agent-attended.ts

```diff
@@ -0,0 +1,47 @@
+import {hl7V3} from "@models"
+
+import pino from "pino"
+
+import {createRepresentedOrganization} from "./agent-unattended"
+
+export async function createAuthorForAttendedAccess(
+  professionalCode: string,
+  logger: pino.Logger,
+  organizationCode?: string,
+): Promise<hl7V3.Author> {
+  const agentPerson = await createAgentPersonForAttendedAccess(professionalCode, logger, organizationCode)
+  const author = new hl7V3.Author()
+  author.AgentPerson = agentPerson
+  return author
+}
+
+export async function createAgentPersonForAttendedAccess(
+  professionalCode: string,
+  logger: pino.Logger,
+  organizationCode?: string,
+): Promise<hl7V3.AgentPerson> {
+  const agentPerson = new hl7V3.AgentPerson()
+  agentPerson.id = new hl7V3.UnattendedSdsRoleProfileIdentifier()
+  agentPerson.code = new hl7V3.UnattendedSdsJobRoleCode()
+  const telecom = new hl7V3.Telecom()
+  telecom._attributes = {
+    use: hl7V3.TelecomUse.WORKPLACE,
+    value: "tel:01234567890"
+  }
```

Comment:
Again this might be a project wide choice but if we had optional arguements in a lot of these class constructors we could avoid having to do these manual reassigns

Reply:
i like the clarity it provides tbh, putting a field in a constructor hides which part of the class is being populated